Week3/homework/todoApp.sql

```diff
@@ -0,0 +1,121 @@
+
+DROP SCHEMA IF EXISTS todoapp;
+CREATE SCHEMA todoapp;
+USE todoapp;
+
+DROP TABLE IF EXISTS users;
+
+CREATE TABLE users
+(
+  `user_id` varchar
+(50) NOT NULL,
+  `name` varchar
+(50) NOT NULL,
+  `email` varchar
+(254) NOT NULL,
+  PRIMARY KEY
+(`user_id`)
+) ;
+
+
+LOCK TABLES `users` WRITE;
+INSERT INTO 
+users
+VALUES
+  ('8vS4PDKvV-aXUgnSu1I4B', 'jack', 'jack@gmail.com'),
+  ('8wuS06DCjCTqv8AD6qTUX', 'mike', 'mike@hotmail.com'),
+  ('HCN4HAjIj3lIKB9tb9doc', 'noer', 'noer@gmail.com'),
+  ('Hk0yPR8DvmIC7WYOvDkgP', 'mohammad', 'mohammad@gmail.com'),
+  ('KHuSxKyVy6AGwuIW3-o1y', 'ahmad', 'ahmad@gmail.com'),
+  ('RfVkWYLz7t9xWnECw8zfo', 'ammar', 'ammar@gmail.com'),
+  ('sPcDYztwcEjYTEIYLP8h3', 'omar', 'omar@gmail.com'),
+  ('tIzOGGdf-95KfrApfy6yq', 'marko', 'marko@hotmail.com'),
+  ('Vn9vc1d5XFxE4T4B0PVGe', 'bert', 'bert@kgmail.com'),
+  ('ySEt2NHE0HvW0sAtX97mK', 'marion', 'marion@hotmail.com');
+UNLOCK TABLES;
+
+
+
+DROP TABLE IF EXISTS todolists;
+
+CREATE TABLE todolists
+(
+  `user_id` varchar
+(50) NOT NULL,
+  `todolist_id` varchar
+(50) NOT NULL,
+  `todolist_name` varchar
+(50) NOT NULL,
+  `reminder` datetime DEFAULT NULL,
+  PRIMARY KEY
+(`todolist_id`),
+  FOREIGN KEY
+(`user_id`) REFERENCES `users`
+(`user_id`) ON
+DELETE CASCADE
+);
+
+LOCK TABLES `todolists` WRITE;
+INSERT INTO 
+todolists
+VALUES
+  ('RfVkWYLz7t9xWnECw8zfo', '-gEwx6en9qLEr0ZRuE5Q-', 'cleaning material', '2019-11-03 00:00:00'),
+  ('Hk0yPR8DvmIC7WYOvDkgP', '0U8YAoKhi31TGu9BHPYjt', 'furniture', '2019-10-02 00:00:00'),
+  ('Vn9vc1d5XFxE4T4B0PVGe', '3UthG7gWxIAcI-WbXAM9j', 'grocery1', '2019-05-29 00:00:00'),
+  ('ySEt2NHE0HvW0sAtX97mK', '6oWRF5tlAh234uyPUHn76', 'house stuff 123', '2019-07-09 00:00:00'),
+  ('8vS4PDKvV-aXUgnSu1I4B', 'BvDSlKNJeadmhjtyANlBG', 'library', '2019-04-02 00:00:00'),
+  ('Vn9vc1d5XFxE4T4B0PVGe', 'iLI-Z83ooba6RQAglkwyk', 'laundry', '2019-05-02 00:00:00'),
+  ('KHuSxKyVy6AGwuIW3-o1y', 'Lr8AxOBnJ6U_j7XiAl4Vq', 'fruits', '2019-10-02 00:00:00'),
+  ('sPcDYztwcEjYTEIYLP8h3', 'lZSG7Ww5h84e9J5LOxe1T', 'digital stuff', '2019-10-02 00:00:00'),
+  ('HCN4HAjIj3lIKB9tb9doc', 'MsOjgJCz0KuUQ9z_4NTUu', 'language', '2019-09-02 00:00:00'),
+  ('8wuS06DCjCTqv8AD6qTUX', 'NEeJQhlvb7r4O3hMsd7CI', 'housing', '2019-04-02 00:00:00'),
+  ('8wuS06DCjCTqv8AD6qTUX', 'PKcCDX58hzwDZAv3ETGzC', 'course', '2019-04-02 00:00:00'),
+  ('tIzOGGdf-95KfrApfy6yq', 'sJDNvkiIpkARhvdkC_sMt', 'meetings', '2019-04-09 00:00:00');
+UNLOCK TABLES;
+
+DROP TABLE IF EXISTS todos;
+
+CREATE TABLE todos
+(
+  `todolist_id` varchar
+(50) NOT NULL,
+  `todo_id` varchar
+(50) NOT NULL,
+  `todo_name` varchar
+(100) NOT NULL,
+  `done` enum
+('true','false') DEFAULT 'false',
```

Comment:
why enum and not boolean?

Reply:
Hello @danlaudk ,
I used enum because i want the value to be  `only` true or false ......I think if i used boolean 1 and 0 will considered also as true and false  .....If do you think boolean will be better in this situation please can you explain to me why ? 
Best regards,

Reply:
you're right about 1/0 . i'd question why you don't want to allow 1/0 (ie only true false). 

both js and db natively support boolean : takes less space, is standard way to encode binary value. Unless you expect to add more values to the enum in the future (a 3rd value called "almost done", and then you should probably code that already) ... So mostly it looks really weird (because it's non standard), making the person reading your code scratch their head and ask "why not boolean"?